modules/command-engine/core/src/main/scala/surge/core/KafkaProducerActor.scala

```diff
@@ -58,6 +59,59 @@ object KafkaProducerActor {
       signalBus)
   }
 
+  def create(
+      actorSystem: ActorSystem,
+      metrics: Metrics,
+      businessLogic: SurgeModel[_, _, _, _],
+      kStreams: AggregateStateStoreKafkaStreams[_],
+      partitionTracker: KafkaConsumerPartitionAssignmentTracker,
+      signalBus: HealthSignalBusTrait,
+      config: Config,
+      kafkaProducerOverride: Option[KafkaProducerTrait[String, Array[Byte]]] = None): Int => KafkaProducerActor = partitionNumber => {
```

Comment:
`create` is a little confusing for a name here since this returns a function that creates producer actors based on a partition number rather than a producer actor directly. Maybe something like `createFromPartitionFunction`? I'm open to better suggestions as well :)